### DISCUSSION TITLE
Is there way to add api prefix  ?

### DISCUSSION THREAD

**Discussion by Guiwoo** [comfyanonymous/ComfyUI#2878] [Q&A]
Hi I would like to set api prefix.
currently comfyUI server running as charm and I can call like localhost:8188/history or prompt 
I would like to set localhost:8188/prefix/history like this.
Is there a way to set prefix once ? I do not want to find all route path and add this prefix

**Comment by hydrusbeta** [comment]
I don't see a built-in way to do this right now. However, if you're willing to modify the code a little, you can take advantage of aiohttp's "sub application" feature.

Make the following modifications to server.py:
1. in PromptServer's constructor, save "middlewares" and "max_upload_size" as attributes. You may also remove these arguments from the Application constructor:
![image](https://github.com/comfyanonymous/ComfyUI/assets/128447736/dff2127b-ecff-48a8-9ab9-39f832f9f76a)

2. in the start() method, instantiate an Application object and register self.app as a sub application with a given prefix. In this example, I'm using the prefix "/prefix/". Pass the parent Application object to the AppRunner.
![image](https://github.com/comfyanonymous/ComfyUI/assets/128447736/3a5bb7fc-4b8c-4880-aa23-e03fc6cc0707)

3. Now start ComfyUI and go to localhost:8188/prefix/
![image](https://github.com/comfyanonymous/ComfyUI/assets/128447736/1caaff35-c654-495e-b58f-7d5f43fc9ae3)
You should find that other URLs work with the prefix too, such as localhost:8188/prefix/history

**Reply by Guiwoo** [reply]
Thanks !
